packages/utils/src/index.ts

```diff
@@ -508,58 +508,48 @@ export const processReferencesForCF = curry(
     }
 
     return fieldsToPopulate.reduce(async (chain, opt) => {
-      const { field, populate } = opt
+      const { field, populate, subFields } = opt
       const val = get(document, field)
 
       const processRefs = processReferencesForCF(firestoreService, {
-        populate: populateAllTheThings ? true : populate
+        populate: populateAllTheThings
+          ? true
+          : Array.isArray(subFields)
+          ? { populate: subFields }
+          : populate
```

Comment:
nah nah nah!!! ternary ternary!!! my braaaaaain

Reply:
ya, haha, don't like nested ternaries, but actually ended up being the easiest fix. Went down the rabbit hole of trying to handle subFields inside each type of value check and then realized I could just handle it recursively if the subfields become the next level's main `populate` fields.